modules/lambda/main.tf

```
@@ -15,37 +13,37 @@ module "label" {
   context = module.this.context
 }
 
-resource "aws_iam_policy" "default" {
-  count = local.iam_policy_enabled ? 1 : 0
+module "iam_policy" {
+  count   = local.iam_policy_enabled ? 1 : 0
+  source  = "cloudposse/iam-policy/aws"
+  version = "1.0.1"
 
-  name        = module.label.id
-  path        = "/"
-  description = format("%s Lambda policy", module.label.id)
-  policy      = var.policy_json
+  iam_policy_enabled          = true
+  iam_policy                  = var.iam_policy
+  iam_source_policy_documents = [var.policy_json]
```

While this works out to be OK in this case with the current implementation of `iam-policy`, Cloud Posse style guidelines require that lists used as inputs to modules have lengths known at plan time and no empty elements. This is to enable `count` and `for_each` to work.

```suggestion
  iam_source_policy_documents = var.policy_json != null ? [var.policy_json] : []
```